std/datetime/systime.d

```diff
@@ -8520,19 +8519,18 @@ public:
         if (isSomeString!(S))
     {
         import std.algorithm.searching : countUntil, find;
-        import std.conv : to;
         import std.string : strip;
 
-        auto dstr = to!dstring(strip(isoExtString));
+        auto str = strip(isoExtString);
 
-        auto tIndex = dstr.countUntil('T');
+        auto tIndex = str.countUntil('T');
         enforce(tIndex != -1, new DateTimeException(format("Invalid ISO Extended String: %s", isoExtString)));
 
-        auto found = dstr[tIndex + 1 .. $].find('.', 'Z', '+', '-');
-        auto dateTimeStr = dstr[0 .. $ - found[0].length];
+        auto found = str[tIndex + 1 .. $].find('.', 'Z', '+', '-');
```

Comment:
Even without checking for Unicode validity, bad Unicode and valid, non-ASCII characters would fail and result in a `DateTimeException` if no auto-decoding happens and thows a `UTFException`, because the code units would not be valid in the ISO format.

Regardless, for this to work correctly, `countUntil` is going to need to calculate the index correctly, which it won't if you're not dealing with code points that are a single code unit. I'd suggest switching to using `indexOf` instead, but it has this rather odd note in its documentation:

        If the parameters are not valid UTF, the result will still
        be in the range [-1 .. s.length], but will not be reliable otherwise.

and I don't know what that really means without taking the time to dig through the source code to figure out what it's really doing.

Alternatively, `find` could be used, in which case the `enforce` would check that the result is non-empty, and the code after it would need to be adjusted given that you don't directly have the index anymore.